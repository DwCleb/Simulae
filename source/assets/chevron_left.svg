<svg width="12" height="18" viewBox="0 0 12 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M11.1328 2.10938L4.24219 9L11.1328 15.8906L9.02344 18L0.0234375 9L9.02344 0L11.1328 2.10938Z" fill="#BDBDBD"/>
</svg>
