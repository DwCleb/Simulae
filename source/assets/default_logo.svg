<svg width="463" height="117" viewBox="0 0 463 117" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M30.5914 116.528C25.2154 116.528 20.2234 115.568 15.6154 113.648C11.0074 111.728 7.31138 108.8 4.52738 104.864C1.83938 101.024 0.495375 96.32 0.495375 90.752C0.495375 83.936 2.36737 79.04 6.11137 76.064C9.85538 72.992 14.1274 71.456 18.9274 71.456C16.3354 74.24 14.3674 77.312 13.0234 80.672C11.6794 84.032 11.0074 87.392 11.0074 90.752C11.0074 96.512 12.8794 101.312 16.6234 105.152C20.4634 108.992 25.9354 110.912 33.0394 110.912C37.5514 110.912 41.8234 110.096 45.8554 108.464C49.9834 106.832 53.3434 104.432 55.9354 101.264C58.5274 98.096 59.8234 94.304 59.8234 89.888C59.8234 86.048 58.8634 82.688 56.9434 79.808C55.0234 76.832 52.6234 74.336 49.7434 72.32C46.8634 70.208 43.0234 67.712 38.2234 64.832C33.3274 61.952 29.3914 59.456 26.4154 57.344C23.5354 55.136 21.0874 52.496 19.0714 49.424C17.0554 46.352 16.0474 42.8 16.0474 38.768C16.0474 33.2 17.7754 27.44 21.2314 21.488C24.6874 15.44 29.5354 10.4 35.7754 6.368C42.1114 2.336 49.2154 0.319996 57.0874 0.319996C62.3674 0.319996 66.2074 1.616 68.6074 4.208C71.0074 6.8 72.2074 10.304 72.2074 14.72C72.2074 19.52 71.0554 24.896 68.7514 30.848C66.4474 36.8 63.9034 41.984 61.1194 46.4C57.9514 45.728 56.3674 43.664 56.3674 40.208C56.3674 38.384 56.7034 36.32 57.3754 34.016C58.1434 31.712 59.1994 28.832 60.5434 25.376C61.9834 21.632 63.0394 18.608 63.7114 16.304C64.4794 13.904 64.8634 11.792 64.8634 9.96799C64.8634 7.95199 64.2394 6.41599 62.9914 5.35999C61.8394 4.30399 59.7754 3.77599 56.7994 3.77599C51.1354 3.77599 45.8554 5.55199 40.9594 9.10399C36.1594 12.56 32.3194 16.784 29.4394 21.776C26.5594 26.768 25.1194 31.328 25.1194 35.456C25.1194 38.528 25.9354 41.264 27.5674 43.664C29.2954 46.064 31.4074 48.176 33.9034 50C36.3994 51.728 39.8074 53.792 44.1274 56.192C49.3114 59.168 53.4874 61.856 56.6554 64.256C59.8234 66.56 62.5114 69.536 64.7194 73.184C67.0234 76.736 68.1754 80.96 68.1754 85.856C68.1754 92.192 66.3034 97.664 62.5594 102.272C58.9114 106.88 54.2074 110.384 48.4474 112.784C42.6874 115.28 36.7354 116.528 30.5914 116.528ZM106.211 41.792C104.483 41.792 103.043 41.408 101.891 40.64C100.835 39.776 100.307 38.576 100.307 37.04C100.307 35.12 101.459 33.392 103.763 31.856C106.067 30.32 108.419 29.552 110.819 29.552C113.987 29.552 115.57 31.04 115.57 34.016C115.57 35.84 114.563 37.616 112.547 39.344C110.627 40.976 108.515 41.792 106.211 41.792ZM88.9305 107.744C85.3785 107.744 82.6425 106.592 80.7225 104.288C78.8025 102.08 77.8425 99.2 77.8425 95.648C77.8425 91.904 78.8985 86.72 81.0105 80.096C83.2185 73.472 85.9065 67.664 89.0745 62.672C92.3385 57.68 95.5065 55.184 98.5785 55.184C99.4425 55.184 100.163 55.472 100.739 56.048C101.411 56.624 101.747 57.392 101.747 58.352C101.747 59.696 100.211 62.912 97.1385 68C93.9705 73.28 91.4265 78.032 89.5065 82.256C87.6825 86.384 86.7705 90.32 86.7705 94.064C86.7705 97.424 87.2985 99.728 88.3545 100.976C89.5065 102.128 91.3305 102.704 93.8265 102.704C101.315 102.704 108.707 95.36 116.003 80.672L117.299 82.112C114.611 90.176 110.627 96.464 105.347 100.976C100.163 105.488 94.6905 107.744 88.9305 107.744ZM185.873 108.464C182.705 108.464 180.305 107.456 178.673 105.44C177.041 103.52 176.225 101.024 176.225 97.952C176.225 95.36 176.993 91.568 178.529 86.576C179.009 84.944 179.441 83.312 179.825 81.68C180.305 80.048 180.545 78.656 180.545 77.504C180.545 75.776 180.113 74.48 179.249 73.616C178.481 72.752 177.521 72.32 176.369 72.32C174.161 72.32 171.761 73.904 169.169 77.072C166.673 80.144 162.785 85.712 157.505 93.776C156.161 95.888 154.289 99.44 151.889 104.432L150.593 107.168C149.057 107.168 147.425 106.88 145.697 106.304C144.065 105.824 143.249 105.152 143.249 104.288C143.249 103.616 144.065 101.072 145.697 96.656C147.713 90.704 149.297 85.712 150.449 81.68C151.697 77.552 152.321 74.048 152.321 71.168C152.321 67.52 151.121 65.696 148.721 65.696C146.033 65.696 142.529 67.856 138.209 72.176C133.985 76.496 129.809 81.968 125.681 88.592C121.553 95.12 118.337 101.552 116.033 107.888C114.593 107.888 112.961 107.552 111.137 106.88C109.217 106.208 108.257 105.536 108.257 104.864C108.257 104.576 108.929 102.32 110.273 98.096C112.481 91.664 114.305 85.76 115.745 80.384C117.281 75.008 118.097 70.352 118.193 66.416C118.865 65.648 119.825 64.976 121.073 64.4C122.321 63.728 123.569 63.392 124.817 63.392C127.409 63.392 128.705 64.64 128.705 67.136C128.705 69.728 127.457 75.2 124.961 83.552C129.665 76.064 134.705 69.632 140.081 64.256C145.553 58.88 150.641 56.192 155.345 56.192C158.129 56.192 160.097 57.2 161.249 59.216C162.497 61.232 163.121 63.872 163.121 67.136C163.121 70.4 162.641 73.952 161.681 77.792C160.721 81.632 159.713 84.848 158.657 87.44C162.305 80.816 166.097 75.488 170.033 71.456C173.969 67.328 178.145 65.264 182.561 65.264C185.441 65.264 187.505 66.128 188.753 67.856C190.097 69.584 190.769 71.792 190.769 74.48C190.769 76.304 190.529 78.272 190.049 80.384C189.569 82.4 188.849 84.896 187.889 87.872C186.257 92.96 185.441 96.512 185.441 98.528C185.441 101.792 186.881 103.424 189.761 103.424C193.313 103.424 196.673 101.264 199.841 96.944C203.009 92.528 206.081 87.104 209.057 80.672L210.785 82.688C208.097 89.792 204.593 95.888 200.273 100.976C195.953 105.968 191.153 108.464 185.873 108.464ZM217.191 108.32C214.023 108.32 211.431 106.88 209.415 104C207.399 101.216 206.391 97.808 206.391 93.776C206.391 90.224 207.111 85.856 208.551 80.672C210.087 75.392 211.863 70.832 213.879 66.992C215.991 63.056 217.719 61.088 219.063 61.088C219.639 61.088 220.455 61.376 221.511 61.952C222.663 62.528 223.623 63.2 224.391 63.968C222.375 67.136 220.263 71.696 218.055 77.648C215.847 83.504 214.743 88.592 214.743 92.912C214.743 95.12 215.079 97.28 215.751 99.392C216.519 101.504 217.911 102.56 219.927 102.56C223.287 102.56 226.935 99.488 230.871 93.344C234.903 87.104 239.223 79.232 243.831 69.728L246.135 64.976C250.743 65.84 253.047 67.088 253.047 68.72C253.047 68.912 252.807 69.584 252.327 70.736C251.943 71.792 251.559 72.752 251.175 73.616C248.967 78.512 247.095 83.12 245.559 87.44C244.119 91.664 243.399 95.264 243.399 98.24C243.399 102.08 244.983 104 248.151 104C250.263 104 252.759 103.088 255.639 101.264C258.519 99.344 261.447 96.608 264.423 93.056C267.399 89.504 269.991 85.376 272.199 80.672L274.071 81.968C272.055 87.056 269.415 91.616 266.151 95.648C262.983 99.584 259.527 102.656 255.783 104.864C252.039 107.168 248.487 108.32 245.127 108.32C238.407 108.32 235.047 104.768 235.047 97.664C235.047 95.744 235.383 93.344 236.055 90.464C232.983 95.936 229.911 100.304 226.839 103.568C223.767 106.736 220.551 108.32 217.191 108.32ZM281.483 108.32C277.547 108.32 274.619 106.976 272.699 104.288C270.875 101.6 269.963 98 269.963 93.488C269.963 84.848 272.939 73.136 278.891 58.352C284.843 43.472 291.467 30.08 298.763 18.176C306.155 6.272 311.483 0.319996 314.747 0.319996C316.091 0.319996 317.195 1.51999 318.059 3.91999C319.019 6.224 319.499 8.576 319.499 10.976C319.499 15.392 317.675 21.92 314.027 30.56C310.475 39.104 305.771 47.792 299.915 56.624C294.059 65.456 287.819 72.608 281.195 78.08C279.947 83.648 279.323 88.496 279.323 92.624C279.323 100.016 281.435 103.712 285.659 103.712C289.211 103.712 292.811 101.456 296.459 96.944C300.203 92.432 303.515 87.008 306.395 80.672L308.555 81.824C300.299 99.488 291.275 108.32 281.483 108.32ZM282.491 73.184C287.771 67.904 292.907 61.28 297.899 53.312C302.987 45.248 307.115 37.568 310.283 30.272C313.451 22.88 315.035 17.552 315.035 14.288C315.035 12.752 314.699 11.984 314.027 11.984C312.683 11.984 309.899 15.296 305.675 21.92C301.547 28.448 297.275 36.464 292.859 45.968C288.443 55.472 284.987 64.544 282.491 73.184ZM310.475 110.768C307.307 110.768 304.763 109.712 302.842 107.6C300.826 105.584 299.818 102.656 299.818 98.816C299.818 93.152 301.834 86.816 305.867 79.808C309.995 72.8 315.179 66.848 321.419 61.952C327.659 57.056 333.755 54.608 339.707 54.608C343.451 54.608 346.571 55.616 349.067 57.632C351.563 59.648 352.811 62.24 352.811 65.408C352.811 67.616 352.187 69.2 350.939 70.16C349.691 71.12 347.963 71.6 345.755 71.6C346.235 69.872 346.475 68.192 346.475 66.56C346.475 64.16 345.899 62.24 344.747 60.8C343.595 59.264 341.819 58.496 339.419 58.496C335.195 58.496 330.731 60.848 326.027 65.552C321.419 70.256 317.531 75.872 314.363 82.4C311.291 88.928 309.755 94.544 309.755 99.248C309.755 102.992 311.051 104.864 313.643 104.864C317.483 104.864 321.995 101.84 327.179 95.792C332.363 89.648 337.931 82.208 343.883 73.472C345.131 73.664 346.379 74.096 347.627 74.768C348.971 75.344 349.643 76.112 349.643 77.072C349.643 77.84 349.163 79.088 348.203 80.816C347.339 82.448 346.811 83.408 346.619 83.696C344.987 86.48 343.691 88.976 342.731 91.184C341.771 93.296 341.291 95.264 341.291 97.088C341.291 98.72 341.675 100.208 342.443 101.552C343.307 102.8 344.555 103.424 346.187 103.424C348.971 103.424 352.475 101.456 356.699 97.52C360.923 93.488 365.099 87.872 369.227 80.672L370.667 82.112C368.651 87.2 366.059 91.76 362.891 95.792C359.723 99.728 356.315 102.8 352.667 105.008C349.019 107.216 345.515 108.32 342.155 108.32C338.795 108.32 336.203 107.312 334.379 105.296C332.651 103.376 331.787 101.024 331.787 98.24C331.787 96.992 331.931 95.6 332.219 94.064C327.899 99.824 324.011 104.048 320.555 106.736C317.195 109.424 313.835 110.768 310.475 110.768ZM381.833 110.336C376.169 110.336 371.801 108.752 368.729 105.584C365.753 102.512 364.265 98.528 364.265 93.632C364.265 87.968 366.137 81.872 369.881 75.344C373.625 68.816 378.329 63.344 383.993 58.928C389.753 54.416 395.369 52.16 400.841 52.16C406.793 52.16 409.769 55.088 409.769 60.944C409.769 65.648 407.993 70.064 404.441 74.192C400.889 78.224 396.329 81.536 390.761 84.128C385.193 86.72 379.625 88.304 374.057 88.88C373.673 90.512 373.481 92.24 373.481 94.064C373.481 97.52 374.345 100.4 376.073 102.704C377.801 105.008 380.729 106.16 384.857 106.16C390.713 106.16 396.617 103.712 402.569 98.816C408.521 93.92 413.369 87.872 417.113 80.672L418.841 81.968C415.001 90.512 409.673 97.376 402.857 102.56C396.041 107.744 389.033 110.336 381.833 110.336ZM374.777 86.144C378.809 84.704 383.033 82.688 387.449 80.096C391.865 77.504 395.609 74.48 398.681 71.024C401.849 67.472 403.433 63.776 403.433 59.936C403.433 58.784 403.193 57.968 402.713 57.488C402.329 56.912 401.561 56.624 400.409 56.624C397.433 56.624 394.169 58.112 390.617 61.088C387.161 64.064 383.945 67.856 380.969 72.464C378.089 77.072 376.025 81.632 374.777 86.144ZM418.121 44.96C416.873 42.464 415.433 40.256 413.801 38.336C412.169 36.416 410.057 34.208 407.465 31.712C404.201 32.864 401.417 34.064 399.113 35.312C396.809 36.56 394.649 38.096 392.633 39.92C391.673 39.344 391.193 38.432 391.193 37.184C391.193 35.456 392.249 33.632 394.361 31.712C396.473 29.696 398.873 28.016 401.561 26.672C404.249 25.232 406.265 24.512 407.609 24.512C409.145 24.512 410.921 25.424 412.937 27.248C415.049 29.072 416.825 31.232 418.265 33.728C419.801 36.224 420.569 38.48 420.569 40.496C420.569 42.608 419.753 44.096 418.121 44.96ZM419.994 89.168C419.994 86.192 420.714 82.448 422.154 77.936C423.594 73.424 425.514 68.048 427.914 61.808C435.402 42.416 441.642 27.344 446.634 16.592C451.722 5.74399 455.994 0.319996 459.45 0.319996C460.218 0.319996 460.89 0.655994 461.466 1.32799C462.042 1.904 462.33 2.672 462.33 3.632C462.33 6.128 459.594 13.232 454.122 24.944C448.746 36.656 442.602 49.04 435.69 62.096C428.778 75.056 423.546 84.08 419.994 89.168ZM415.818 105.44C414.474 105.44 413.37 105.104 412.506 104.432C411.642 103.76 411.21 102.8 411.21 101.552C411.21 100.016 412.122 98.672 413.946 97.52C415.77 96.272 417.642 95.648 419.562 95.648C422.058 95.648 423.306 96.8 423.306 99.104C423.306 100.544 422.49 101.984 420.858 103.424C419.322 104.768 417.642 105.44 415.818 105.44Z" fill="#F2F2F2"/>
</svg>
