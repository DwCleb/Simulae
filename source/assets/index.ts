export { default as Logo } from './default_logo.svg';
export { default as BackIcon } from './chevron_left.svg';
export { default as CleberAvatar } from './cleber_avatar.jpg';
