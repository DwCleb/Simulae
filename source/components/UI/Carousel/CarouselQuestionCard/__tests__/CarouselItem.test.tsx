import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import CarouselItem from '../index';
import { CarouselItemProps } from '../index';

import { Logo } from '~/assets';

import darkTheme from '~/theme/ThemeContext/dark';

let wrapper = {};

const props: CarouselItemProps = { item: { title: '', description: '', image: Logo, overlap: false } };

const theme = darkTheme.theme;

describe('<CarouselItem />', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  beforeEach(() => {
    wrapper = create(
      <ThemeProvider theme={theme}>
        <CarouselItem {...props} />
      </ThemeProvider>
    );
    return wrapper;
  });

  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
