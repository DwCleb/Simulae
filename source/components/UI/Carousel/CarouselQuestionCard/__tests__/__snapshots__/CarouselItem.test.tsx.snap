// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselItem /> Should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 300,
        "marginTop": 30,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      align={true}
      distance={
        Array [
          0,
          40,
          0,
          40,
        ]
      }
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Jawwy-Medium",
            "fontSize": 48,
            "letterSpacing": 0,
            "lineHeight": 72,
            "paddingBottom": 0,
            "paddingLeft": 80,
            "paddingRight": 80,
            "paddingTop": 0,
            "textAlign": "center",
            "textTransform": "none",
          },
        ]
      }
      type="H3"
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 3,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      height="100%"
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/jawwy_logo.png",
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": "100%",
            "marginBottom": 100,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 100,
            "width": "100%",
          },
          Object {},
        ]
      }
      width="100%"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      align={true}
      distance={
        Array [
          0,
          48,
          0,
          48,
        ]
      }
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Jawwy-Regular",
            "fontSize": 32,
            "letterSpacing": 0,
            "lineHeight": 40,
            "paddingBottom": 0,
            "paddingLeft": 96,
            "paddingRight": 96,
            "paddingTop": 0,
            "textAlign": "center",
            "textTransform": "none",
          },
        ]
      }
      type="P1"
    >
      
    </Text>
  </View>
</View>
`;
